Don't print 'convenience' on protocol extension initializers

jrose-apple opened the pull request:
We model them that way in the compiler, but that's not part of the user-level language.

rdar://problem/32067077

jrose-apple commented:
@swift-ci Please test

nkcsgexi reviewed (approved):
Thank you for fixing this!

jrose-apple commented:
@swift-ci Please test Linux

rintaro commented on lib/AST/ASTPrinter.cpp:
```diff
@@ -2679,11 +2679,14 @@ void PrintAST::visitConstructorDecl(ConstructorDecl *decl) {
        decl->getInitKind() == CtorInitializerKind::ConvenienceFactory) &&
-      !decl->getAttrs().hasAttribute<ConvenienceAttr>()) {
+      !decl->getAttrs().hasAttribute<ConvenienceAttr>() &&
+      !decl->getDeclContext()->getAsProtocolExtensionContext()) {
```
How about `class`? Inherited `init` from protocol extension in `class` should be printed as `convenience`.
IIRC, we don't have test case for that. https://github.com/apple/swift/pull/11099#issuecomment-317303609

jrose-apple replied:
Hm, I guess so. Okay, I'll check the effective context, or whatever it's called.

jrose-apple commented:
@swift-ci Please smoke test

jrose-apple committed:
Don't print 'convenience' on protocol extension initializers

We model them that way in the compiler, but that's not part of the
user-level language.

rdar://problem/32067077

jrose-apple commented:
@swift-ci Please smoke test

rintaro reviewed (approved):
Thank you!
